lib/python/treadmill_aws/garbage_collector.py

```diff
@@ -0,0 +1,75 @@
+"""AWS specific treadmill garbage collector
+"""
+
+import logging
+from datetime import datetime
+from datetime import timedelta
+
+import jmespath
+
+from treadmill import admin
+from treadmill import context
+from treadmill_aws.awscontext import GLOBAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _valid_ec2_instances():
+    try:
```

Comment:
This cache is never unset- would this mean that the list of instances is not refreshed once set?